subsys/net/ip/tcp.c

```diff
@@ -417,7 +417,9 @@ int net_tcp_prepare_segment(struct net_tcp *tcp, u8_t flags,
 
 	if (flags & NET_TCP_FIN) {
 		tcp->flags |= NET_TCP_FINAL_SENT;
-		seq++;
+printk("Making sure FIN goes with ACK; adjusting seq by 2; now fire up Wireshark and check that all's ok\n");
```

Comment:
It's on purpose, as mentioned in the comment:

> A stray printk() should help avoid premature merge.

As https://jira.zephyrproject.org/browse/ZEP-2104 alone didn't catch attention, maybe this draft patch will.